2016-12-3_Git-and-Github-Best-Practices.md

```diff
@@ -0,0 +1,182 @@
+
+Look at this commit message:
+
+![A good commit](github_commit-good-example.png)
+
+From this commit you get a short, descriptive summary in the proper tense, and an extended description that lets you know exactly what has changed without needing to open the files.  Most importantly, it clearly shows **why** the commit exists in the first place:
+
+> Schema currently points to outdated href, this patch fixes the attribute to reference the proper schema.
+
+In conclusion, each commit gives a small summary, but also a longer description that fully document the changes made.  This description is key in tracking where issues come from and **_why_** someone made certain changes or decisions.
+
+![Github Commit dialog](github_commit-upload.png)
+
+Make sure to write custom commit messages on GitHub as well!  Leaving the default message clutters the [history](https://github.com/dig-eg-gaz/content/commits/master) with messages that don't offer any insight into changes made.
+
+## Issues
+
+Issues ideally follow the same convention for commit summaries; provide a description in the present imperative tense.  However, issues are far more flexible; generally, each project has its own conventions for issues that the contributors follow.
+
+Some universial best practices are as follows:
+
+- Provide as [much information as necessary](https://github.com/dig-eg-gaz/advertisements/issues/6) (or possible) to solve the issue
+- Check to see if duplicates exist first
+- Reference any relevant commits or pull requests
+
+Creating good issues can actually help open-source projects immensely.
+
+Many projects use [lables](https://github.com/dig-eg-gaz/content/labels) to organize issues, especially once a project starts to accumulate a lot of issues.  [This project](https://github.com/nasa/openmct/issues) has over 300 issues, labels help them understand which issues are most important.
+
+![Referencing an Issue](github_issue-reference.png)
+
+Furthermore, it's possible to reference issues by their number!  Simply writing `#116`, for example, will add a comment to the issue like the one above, indicating that a reference was made to that issue, allowing for anyone following that issue to see the new one.
+
+## Pull Requests
+
+Pull requests are one of the most important parts of GitHubs web interface.  Really, it doesn't do anything special behind the scenes.  It would be roughly the same as using `git merge` or `git rebase` through the command-line; what it does allow for, however, is centralized review.
+
+### Referencing
+
+Pull Requests can [reference](https://github.com/nasa/openmct/pull/1335) issues, showing a clear connection between issues and pull requests, just like issues can.
+
+![Referencing an issue or pull request in GitHub](github_reference.png)
+
+Always make appropriate references!  This makes sure that the people that relevant information goes where it needs to go.
```

Comment:
Typo: please revise  `This makes sure that the people that relevant information goes where it needs to go.`  First clause is unclear.